src/main/java/org/gwtproject/event/shared/GwtEvent.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2009 Google Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.gwtproject.event.shared;
+
+/**
+ * Root of all GWT widget and dom events sourced by a {@link HandlerManager}.
+ * All GWT events are considered dead and should no longer be accessed once the
+ * {@link HandlerManager} which originally fired the event finishes with it.
+ * That is, don't hold on to event objects outside of your handler methods.
+ * <p>
+ * There is no need for an application's custom event types to extend GwtEvent.
+ * Prefer {@link Event} instead.
+ * 
+ * @param <H> handler type
+ */
+public abstract class GwtEvent<H extends EventHandler> extends Event<H> {
+  /**
+   * Type class used to register events with the {@link HandlerManager}.
+   * <p>
+   * Type is parameterized by the handler type in order to make the addHandler
+   * method type safe.
+   * </p>
+   * 
+   * @param <H> handler type
+   */
+  public static class Type<H> extends Event.Type<H> {
```

Comment:
That type was only kept for backwards compatibility in GWT2. Given that we're killing backwards compat' by changing the package name, better not bring it here.